### PR TITLE
Back port .auto control packages

### PR DESCRIPTION
**Pull request by TakaHoribe**
## Description

[TODO]

- [x] remove autowre_auto_msgs pkg from this PR

Back port from autoware.auto for temporary use

**ported packages:**
 - `control/trajectory_follower`
 - `control/trajectory_follower_nodes`
 -  `simulation/simple_planing_simulator`
 - `common/math/osqp_interface`
 - other dependencies (put under `tmp_autoware_auto_dependencies` directory):
     - `autoware_auto_cmake`
     - `autoware_auto_common`
     - `autoware_auto_geometry`
     - `autoware_auto_tf2`
     - `autoware_testing`
     - `fake_test_node`
     - `motion_common`
     - `motion_testing`
     - `time_utils`

**removed packages**
 - `mpc_follower`
 - `velocity_controller`
 - `simple_planning_simulator`
 - `osqp_interface`

**Base branch**
 -  https://gitlab.com/autowarefoundation/autoware.auto/AutowareAuto/-/merge_requests/1246

**Contained MRs:**
 - https://gitlab.com/autowarefoundation/autoware.auto/AutowareAuto/-/merge_requests/1216
 - https://gitlab.com/autowarefoundation/autoware.auto/AutowareAuto/-/merge_requests/1271

**Applied modification:**
 - galactic support for MR-1216
 - trajectory message modification for MR-1216
 - add missing functions in osqp_interface
 - apply recent autoware_auto_perception_msgs changes

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [ ] Read [pull request guidelines][pull-request-guidelines]
- [ ] Code follows [coding guidelines][coding-guidelines]
- [ ] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] Code follows [coding guidelines][coding-guidelines]
- [ ] (Optional) Unit tests have been written for new behavior
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets
- [ ] Write [release notes][release-notes]

## CI Checks

- **Build and test for PR / build-and-test-pr**: Required to pass before the merge.
- **Build and test for PR / clang-tidy-pr**: NOT required to pass before the merge. It is up to the reviewer(s). Found false positives? See the [guidelines][clang-tidy-guidelines].
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[clang-tidy-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/ClangTidyGuideline/
[coding-guidelines]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/1194394777/T4
[pull-request-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/PullRequestGuideline/
[release-notes]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/563774416
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

**Comment by TakaHoribe** [comment]
The `tmp_autoawre_auto_dependencies` depends on the `autoware_auto_msgs` pkg. @wep21 is working on reviving the pkg. Modules in this PR can be built after the change.

**Comment by TakaHoribe** [comment]
Split message change was applied to this PR. Ready for review

**Comment by tkimura4** [comment]
@kenji-miyake 
autoware.autoから持ってきたパッケージについては, syncが難しくなるためpre-commitを適用しない, としたいのですが可能でしょうか & そもそもその方針で良いでしょうか。

(trajectory_followerなど, tierIV側でも今後それなりに変更が加わりそうなパッケージについては "pre-commitを適用しない"でいいのかどうか...。)

**Comment by tkimura4** [comment]
ignore lint of the packages from autoware.auto.

**Comment by TakaHoribe** [comment]
>autoware.autoから持ってきたパッケージについては, syncが難しくなるためpre-commitを適用しない, としたいのですが可能でしょうか & そもそもその方針で良いでしょうか。

この方針でokとのことで合意済み。

**Comment by tkimura4** [comment]
> : -- run_test.py: invoking following command in '/__w/AutowareArchitectureProposal.iv/AutowareArchitectureProposal.iv/build/motion_common':
1:  - /__w/AutowareArchitectureProposal.iv/AutowareArchitectureProposal.iv/build/motion_common/motion_common_unit_tests --gtest_output=xml:/__w/AutowareArchitectureProposal.iv/AutowareArchitectureProposal.iv/build/motion_common/test_results/motion_common/motion_common_unit_tests.gtest.xml
1: Running main() from /opt/ros/galactic/src/gtest_vendor/src/gtest_main.cc
1: [==========] Running 16 tests from 4 test suites.
1: [----------] Global test environment set-up.
1: [----------] 1 test from HeadingFuncs
1: [ RUN      ] HeadingFuncs.FromQuat
1: [       OK ] HeadingFuncs.FromQuat (0 ms)
1: [----------] 1 test from HeadingFuncs (0 ms total)
1: 
1: [----------] 6 tests from Interpolation
1: [ RUN      ] Interpolation.Clamp
1: [       OK ] Interpolation.Clamp (0 ms)
1: [ RUN      ] Interpolation.Interpolation
1: [       OK ] Interpolation.Interpolation (0 ms)
1: [ RUN      ] Interpolation.Slerp2d
1: /opt/ros/galactic/include/apex_test_tools/apex_test_tools.hpp:81: Failure
1: Failed
1: apex_test_tools::memory_test: Unexpected malloc
1: Stack trace (most recent call last):
1: #18   Object "[0xffffffffffffffff]", at 0xffffffffffffffff, in 
1: #17   Object "/__w/AutowareArchitectureProposal.iv/AutowareArchitectureProposal.iv/build/motion_common/motion_common_unit_tests", at 0x560aba18ba0d, in 
1: #16   Object "/lib/x86_64-linux-gnu/libc.so.6", at 0x7f836e9f60b2, in __libc_start_main
1: #15   Object "/__w/AutowareArchitectureProposal.iv/AutowareArchitectureProposal.iv/build/motion_common/motion_common_unit_tests", at 0x560aba18b5f3, in 
1: #14   Object "/__w/AutowareArchitectureProposal.iv/AutowareArchitectureProposal.iv/build/motion_common/motion_common_unit_tests", at 0x560aba1caa07, in 
1: #13   Object "/__w/AutowareArchitectureProposal.iv/AutowareArchitectureProposal.iv/build/motion_common/motion_common_unit_tests", at 0x560aba1ca7a2, in 
1: #12   Object "/__w/AutowareArchitectureProposal.iv/AutowareArchitectureProposal.iv/build/motion_common/motion_common_unit_tests", at 0x560aba1ca1ec, in 
1: #11   Object "/__w/AutowareArchitectureProposal.iv/AutowareArchitectureProposal.iv/build/motion_common/motion_common_unit_tests", at 0x560aba1ca0c4, in 
1: #10   Object "/__w/AutowareArchitectureProposal.iv/AutowareArchitectureProposal.iv/build/motion_common/motion_common_unit_tests", at 0x560aba1c9f65, in 
1: #9    Object "/__w/AutowareArchitectureProposal.iv/AutowareArchitectureProposal.iv/build/motion_common/motion_common_unit_tests", at 0x560aba1d5d70, in 
1: #8    Object "/__w/AutowareArchitectureProposal.iv/AutowareArchitectureProposal.iv/build/motion_common/motion_common_unit_tests", at 0x560aba18e416, in 
1: #7    Object "/__w/AutowareArchitectureProposal.iv/AutowareArchitectureProposal.iv/build/motion_common/motion_common_unit_tests", at 0x560aba18cc13, in 
1: #6    Object "/__w/AutowareArchitectureProposal.iv/AutowareArchitectureProposal.iv/build/motion_common/motion_common_unit_tests", at 0x560aba197a6e, in 
1: #5    Object "/__w/AutowareArchitectureProposal.iv/AutowareArchitectureProposal.iv/build/motion_common/motion_common_unit_tests", at 0x560aba1ad060, in 
1: #4    Object "/lib/x86_64-linux-gnu/libstdc++.so.6", at 0x7f836edd5b38, in operator new(unsigned long)
1: #3    Object "/opt/ros/galactic/lib/libmemory_tools_interpose.so", at 0x7f836efa17b6, in malloc
1: #2    Object "/opt/ros/galactic/lib/libmemory_tools_interpose.so", at 0x7f836efa1723, in unix_replacement_malloc
1: #1    Object "/opt/ros/galactic/lib/libmemory_tools.so", at 0x7f836ef8c83f, in osrf_testing_tools_cpp::memory_tools::custom_malloc_with_original(unsigned long, void* (*)(unsigned long), char const*, bool)
1: #0    Object "/opt/ros/galactic/lib/libmemory_tools.so", at 0x7f836ef901d9, in void osrf_testing_tools_cpp::memory_tools::print_backtrace<64>(_IO_FILE*)
1:  malloc  (not expected) 392 -> 0x560aba5c20d0
1: /opt/ros/galactic/include/apex_test_tools/apex_test_tools.hpp:81: Failure

**Comment by tkimura4** [comment]
It seems that /opt/ros/galactic/lib/libmemory_tools_interpose.so caused memory leak, then apex_test_tools::memory_test detected it and unit tests failed.

File: tmp_autoware_auto_dependencies/motion_common/test/interpolation.cpp [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+}
+
+TEST(Interpolation, Slerp2d)
+{
+  using motion::motion_common::Orientation;
+  using motion::motion_common::Real;
+  const auto angle_distance = [](Real a, Real b) -> Real
+    {
+      // https://stackoverflow.com/questions/1878907/the-smallest-difference-between-2-angles?rq=1
+      const auto d = a - b;
```
Comment (tkimura4) [inline tmp_autoware_auto_dependencies/motion_common/test/interpolation.cpp:112]: const auto d = b - a

File: tmp_autoware_auto_dependencies/motion_common/test/interpolation.cpp [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+      if (HasFailure()) {
+        const auto ret = slerp(a, b, t);
+        std::cout << "Angles: " << th_a << ", " << th_b << "; " << th_t << ", " << to_angle(ret) <<
+          "\n";
+      }
+    };
+  const auto test_suite =
+    [ = ](Orientation a, Orientation b, Real tol, Real tol2, bool b_snap = false)
+    {
+      constexpr auto lim = std::numeric_limits<Real>::max();
```
Comment (tkimura4) [inline tmp_autoware_auto_dependencies/motion_common/test/interpolation.cpp:145]: lim is 3.4e+38.

It is not appropriate scalar using in slerp function.

**Comment by TakaHoribe** [comment]
@tkimura4 .autoのmasterの時点で同じエラー出てるので、無効化した状態でマージしちゃいましょう。

**Comment by tkimura4** [comment]
@TakaHoribe 
trajectory_follower_nodeの方も無効化でOK?

**Comment by TakaHoribe** [comment]
うん！

**Comment by TakaHoribe** [comment]
master foxy (in ade)

[motion_common-colcon_test-error.txt](https://github.com/tier4/AutowareArchitectureProposal.iv/files/7517081/motion_common-colcon_test-error.txt)

**Review by tkimura4 — APPROVED** [review]
LGTM

**Pull request merged**